pkg/catalogv2/oci/oci.go

```diff
@@ -167,16 +167,17 @@ func GenerateIndex(ociClient *Client, URL string, credentialSecret *corev1.Secre
 		return nil
 	}
 
+	existingCharts := make(map[string]bool)
 	// Loop over all the repositories and fetch the tags
 	repositoriesFunc := func(repositories []string) error {
```

Comment:
```suggestion
	repositoriesFunc := func(repositories []string) error {
	               existingCharts := make(map[string]bool)
```

Reply:
The `repositoriesFunc` is called for every page of results from the _catalog call, so if we declare `existingCharts` inside it, it will reset everytime and won't work for repositories that aren't in the last call